programs/util.c

```diff
@@ -121,6 +121,27 @@ int UTIL_requireUserConfirmation(const char* prompt, const char* abortMsg,
 *  Functions
 ***************************************/
 
+char* humanSize(unsigned long long size, char* str) {
+    if (size > 1125899906842624L) {
+        snprintf(str, 7, "%.1fP", (float)size / 1125899906842624L);
```

Comment:
I see references to `fprintf()`all over the code base so I just assumed `snprintf()` was valid. `util.c.` includes `<stdio.h>` for `fprintf()`, can we rely on that, but not `sprintf()`?

C is not my forte, so I may need some guidance on this one.

Reply:
As C has evolved, they've introduced new functions into the same header. C90 has `fprintf()` but not necessarily `snprintf()`. `snprintf()` was only standardized (and guaranteed to be present) in C99. Zstd restricts itself to the features guaranteed to be present in C90 (or at least has fallbacks to handle that strict case), so that we can be sure it will compile on *all* C90-conforming platforms.